go/teams/seitan.go

```diff
@@ -28,15 +28,26 @@ const SeitanRawIKeyLength = 10
 // is generated on one client and distributed to another via face-to-
 // face meeting, use of a trusted courier etc.
 //
-// We only try to distinguish Seitan tokens from normal e-mail tokens
-// via length so make sure they are never the same length. Right now
-// server-trust e-mail tokens are 12 characters.
-const SeitanEncodedIKeyLength = 16
+// Seitan tokens have a '+' as the fifth character. We use this and length
+// to distinguish from email invite tokens (and team names).
+// Right now server-trust e-mail tokens are 12 characters.
+const SeitanEncodedIKeyLength = 18
+const seitanEncodedIKeyLengthBeforePlus = 17
 
-// Key-Base 32 encoding. lower case letters except 'l', 'i' and digits except for '0' and '1'.
-const KBase32EncodeStd = "abcdefghjkmnopqrstuvwxyz23456789"
+// Key-Base 31 encoding. lower case letters except "lit", and digits except for '0' and '1'.
+const KBase31EncodeStd = "abcdefghjkmnopqrsuvwxyz23456789"
 
-var Base32Encoding = basex.NewEncoding(KBase32EncodeStd, SeitanRawIKeyLength, "")
+var Base31Encoding = basex.NewEncoding(KBase31EncodeStd, SeitanRawIKeyLength, "")
+
+func init() {
+	seitanEncodedIKeyLengthBeforePlus := 17
```

Comment:
Honestly, I think 16 bits would have been OK, but this is fine too. There is nothing magic about 80 bits of entropy pre-stretching. It just seems like it's in the right ballpark.

Reply:
@oconnor663 and I thought basex might panic trying run off the end of output array when calling Encode.. And that wouldn't be exactly 16 bytes of entropy because of the encoding right?

Reply:
Right, the easier way to do this BTW is just:

```golang
var b []byte
for i := 0; i < 16; i++ {
   var r int
   for {
     var tmp [1]byte
     rand.Random(tmp)
     r = tmp[0] & 0x1f
     if r < 31 {
        break
     }
  }
  b = append(b, alphabet[r])
}

```

Reply:
I.e., you'd only have to "base31 encode" if you wanted to encode/decode with fidelity.  But here we have no reason to *decode*.

Reply:
Oh that makes sense. Worth changing here?

Reply:
Oh I see 9808 now